#Meet The Data

The dataset if the well-known Iris dataset. I load it with the load_iris function from the dataset module available in the scikit-learn library.

In [1]:
from sklearn.datasets import load_iris


In [2]:
iris_dataset = load_iris()

The object returned by the load_iris function is called a Bunch object. A Bunch is a Python dictionary that provides attribute-style access (a la JavaScript objects)

#Iris_dataset Keys

In [9]:
print("Keys of iris_dataset: \n{}".format(iris_dataset.keys()))

Keys of iris_dataset: 
dict_keys(['data', 'target', 'target_names', 'DESCR', 'feature_names'])


The key 'DESCR' contains a short description of the dataset. Let's catch a glimpse of it:

In [17]:
print(iris_dataset['DESCR'][:980] + "\n ...")

Iris Plants Database

Notes
-----
Data Set Characteristics:
    :Number of Instances: 150 (50 in each of three classes)
    :Number of Attributes: 4 numeric, predictive attributes and the class
    :Attribute Information:
        - sepal length in cm
        - sepal width in cm
        - petal length in cm
        - petal width in cm
        - class:
                - Iris-Setosa
                - Iris-Versicolour
                - Iris-Virginica
    :Summary Statistics:

    ============== ==== ==== ======= ===== ====================
                    Min  Max   Mean    SD   Class Correlation
    ============== ==== ==== ======= ===== ====================
    sepal length:   4.3  7.9   5.84   0.83    0.7826
    sepal width:    2.0  4.4   3.05   0.43   -0.4194
    petal length:   1.0  6.9   3.76   1.76    0.9490  (high!)
    petal width:    0.1  2.5   1.20  0.76     0.9565  (high!)
    ============== ==== ==== ======= ===== ===================
 ...


The key 'target_names' contains the species of flower that we would like to predict via ML:

In [19]:
print("Target names: \n{}".format(iris_dataset['target_names']))

Target names: 
['setosa' 'versicolor' 'virginica']


Let's check the key that has the features of our dataset:

In [20]:
print("Feature names: \n{}".format(iris_dataset['feature_names']))

Feature names: 
['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


We can take a peek at the measurements of these features:

In [21]:
print("First 10 samples: \n{}".format(iris_dataset['data'][:10]))

First 10 samples: 
[[ 5.1  3.5  1.4  0.2]
 [ 4.9  3.   1.4  0.2]
 [ 4.7  3.2  1.3  0.2]
 [ 4.6  3.1  1.5  0.2]
 [ 5.   3.6  1.4  0.2]
 [ 5.4  3.9  1.7  0.4]
 [ 4.6  3.4  1.4  0.3]
 [ 5.   3.4  1.5  0.2]
 [ 4.4  2.9  1.4  0.2]
 [ 4.9  3.1  1.5  0.1]]


Let's check the dimensions of our 'data' and 'target' keys, that is, how many instances (rows) and attributes (columns) there are:

In [28]:
print("Dimensions of data: \n{}".format(iris_dataset['data'].shape))
print("\n" + "Dimensions of target: \n{}".format(iris_dataset['target'].shape))

Dimensions of data: 
(150, 4)

Dimensions of target: 
(150,)


This means that we have 150 rows (samples of flowers) and 4 columns (features/attributes)

Let's check the data type of the key 'data':

In [24]:
print("Type of data: \n{}".format(type(iris_dataset['data'])))

Type of data: 
<class 'numpy.ndarray'>


This is a multidimensional and homogeneous array of fixed-size items